package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,240 @@
+to perform Principal Component Analysis.
+
+
+.. _PCA-tutorial:
+The example uses files provided as part of the MDAnalysis test suite
+(in the variables :data:`~MDAnalysis.tests.datafiles.PSF` and
+:data:`~MDAnalysis.tests.datafiles.DCD`). This tutorial shows how to use the
+PCA class.
+
+
+First load all modules and test data ::
+    >>> import MDAnalysis as mda
+    >>> import numpy as np
+    >>> import MDAnalysis.analysis.pca as pca
+    >>> from MDAnalysis.tests.datafiles import PSF, DCD
+
+Given a universe containing trajectory data we can perform PCA using
+:class:`PCA`:: and retrieve the principal components.
+    >>> u = mda.Universe(PSF,DCD)
+    >>> PSF_pca = pca.PCA(u)
+    >>> cumulated_variance, principal_components = PSF_pca.fit()
+
+Inspect the components to determine the principal components you would like
+to retain. The choice is arbitrary, but I will stop when 95 percent of the
+variance is explained by the components.
+    >>> n_pcs = next(x[0] for x in enumerate(cumulated_variance) if x[1] > 0.95)
+    >>> pca_space = PSF_pca.transform(n_components=n_pcs)
+
+From here, inspection of the pca_space and conclusions to be drawn from the
+data are left to the user.
+"""
+from six.moves import range
+import logging
+
+import numpy as np
+from MDAnalysis import Universe
+
+from .base import AnalysisBase
+
+logger = logging.getLogger(__name__)
+
+class PCA(AnalysisBase):
+    """Principal component analysis on an MD trajectory
+
+    Attributes
+    ----------
+    p_components: array, (n_components, n_atoms)
+        The principal components of the feature space,
+        representing the directions of maximum variance in the data.
+    variance : array (n_components, )
+        The raw variance explained by each eigenvector of the covariance
+        matrix.
+    explained_variance : array, (n_components, )
+        Percentage of variance explained by each of the selected components.
+        If a subset of components is not chosen then all components are stored
+        and the sum of explained variances is equal to 1.0.
+
+    Methods
+    -------
+    fit(traj=None, select='All', start=None, stop=None, step=None)
+        Find the principal components of selected atoms in a subset of the
+        trajectory.
+    transform(traj=None, n_components=None)
+        Take the original trajectory and project it onto the principal
+        components.
+    inverse_tranform(pca_space)
+        Take a pca_space and map it back onto the trajectory used to create it.
+    """
+
+    def __init__(self, u, select='all'):
+        """
+        Parameters
+        ----------
+        u : MDAnalysis Universe
+            The universe containing the trajectory frames for Principal
+            Component Analysis.
+        """
+        self._u = u
+        self._atoms = self._u.select_atoms(select)
+        self._n_atoms = self._atoms.n_atoms
+        self._calculated = False
+
+    def fit(self, n_components=None, start=None, stop=None,
```

Comment:
Sorry thats a good point. I called it `fit` in an unnecessary attempt to mirror the sklearn api.